signaler/signaler_test.go

```diff
@@ -0,0 +1,37 @@
+//go:build !windows
+// +build !windows
+
+package signaler
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWaitForInterrupt_SIGTERM(t *testing.T) {
+	done := make(chan os.Signal, 1)
+
+	go func() {
+		sig := WaitForInterrupt()
+		done <- sig
+	}()
+
+	// Give the waiter time to register signal.Notify
+	time.Sleep(50 * time.Millisecond)
```

Comment:
🦍 Optional super pedantic nitpick, but the Waiter doesn't need time, it needs a context switch yield.
Better is a way to know the waiter is ready. But that's too complicated to justify changing runtime implementation just to make tests more inspectable for a _very rare_ race.
So my suggestion is a slight change to description, and replacing arbitrary time with specifically yielding.
```suggestion
	// Yield to allow the waiter to register signal.Notify
	runtime.Gosched()
```